pkg/options/types.go

```diff
@@ -106,56 +105,21 @@ func (r *ResourceSet) Type() string {
 }
 
 // NodeType represents a nodeName to query from.
-type NodeType map[string]struct{}
+type NodeType string
 
-// Set converts a comma-separated string of nodename into a slice and appends it to the NodeList
-func (n *NodeType) Set(value string) error {
-	s := *n
-	cols := strings.Split(value, ",")
-	for _, col := range cols {
-		col = strings.TrimSpace(col)
-		if len(col) != 0 {
-			s[col] = struct{}{}
+// GetNodeFieldSelector returns a nodename field selector.
+func (n *NodeType) GetNodeFieldSelector(fetchUnscheduledPods bool) string {
+	if fetchUnscheduledPods {
+		if string(*n) != "" {
+			klog.Warningf("spec.nodeName=%s will not be used, because --enable-unscheduled-pods-fetching is set", string(*n))
 		}
+		klog.InfoS("Using spec.nodeName= to select unscheduable pods without node")
+		return "spec.nodeName="
 	}
```

Comment:
It feels odd to handle this here since fetchUnscheduledPods is not related to the node object. I think we should take care of this scenario in the caller `RunKubeStateMetrics`